mod/blog/classes/Blog/Upgrades/FancyBlogUpgrade2015061700.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Blog\Upgrades;
+
+use Elgg\BatchUpgrade;
+
+class FancyBlogUpgrade2015061700 implements BatchUpgrade {
+
+	private $offset;
+
+	public function getNumRemaining() {
+		$count = elgg_get_entities(array(
+			'type' => 'object',
+			'subtype' => 'blog',
+			'count' => true,
+		));
+
+		return $count - $this->offset;
+	}
+
+	public function isRequired() {
+		return true;
+	}
+
+	public function setOffset($offset) {
```

Comment:
I almost feel like the upgrade class should extend a class instead of implementing the interface. That way methods like this could be implemented in the parent.